experiments/compare_traces.py

```diff
@@ -135,24 +137,25 @@ def all_categoricals_eq(state_traces: List[Tuple[State]]) -> bool:
     # each continuous variable after the completion of each command
     num_commands = len(truth[0].commands)
     size_truth = len(truth)
+    all_vars = SYSTEM.state.variables
     for i in range(num_commands):
         for var in continuous:
             vals = np.array([float(simple_truth[j][i][var])
                              for j in range(size_truth)])
             actual = simple_candidate[i][var]
             mid = max(vals) - ((max(vals) - min(vals)) / 2)
-            tolerance = (max(vals) - min(vals)) / 2
-            tolerance *= tolerance_factor
+#            tolerance = (max(vals) - min(vals)) / 2
+#            tolerance *= tolerance_factor
             diff = abs(mid - actual)
-            is_nearly_eq = np.isclose(mid, actual,
-                                      rtol=1e-05, atol=tolerance, equal_nan=False)
-
+#            is_nearly_eq = np.isclose(mid, actual,
+#                                      rtol=1e-05, atol=tolerance, equal_nan=False)
+            is_nearly_eq = all_vars[var].eq(mid, actual)
```

Comment:
This is a little tricker than what I had in mind, but it might be fine for this particular experiment. It might be worth checking the implementation of the `eq` method to make sure it isn't vulnerable to floating point errors. Perhaps it could be tweaked to use `np.isclose`?

Reply:
How about:
```
is_nearly_eq = np.isclose(mid, actual,
                                      rtol=1e-05, atol=all_vars[var].noise, equal_nan=False)
```
?

Reply:
That's better, but the comparison still looks stricter than it should. Traces within the oracle set might be flagged as inconsistent by this method. Maybe we could add the measurement noise to the threshold?

Reply:
We can either do that or adjust the noise specifications.